tests/test_temporal.py

```diff
@@ -0,0 +1,304 @@
+    )
+
+
+class Deps(TypedDict):
+    country: str
+
+
+async def event_stream_handler(
+    ctx: RunContext[Deps],
+    stream: AsyncIterable[AgentStreamEvent | HandleResponseEvent],
+):
+    logfire.info(f'{ctx.run_step=}')
+    async for event in stream:
+        logfire.info(f'{event=}')
+
+
+async def get_country(ctx: RunContext[Deps]) -> str:
+    return ctx.deps['country']
+
+
+def get_weather(city: str) -> str:
+    return 'sunny'
+
+
+@dataclass
+class Answer:
+    label: str
+    answer: str
+
+
+@dataclass
+class Response:
+    answers: list[Answer]
+
+
+agent = Agent(
+    # Can't use the `openai_api_key` fixture here because the workflow needs to be defined at the top level of the file.
+    OpenAIModel('gpt-4o', provider=OpenAIProvider(api_key=os.getenv('OPENAI_API_KEY', 'mock-api-key'))),
+    deps_type=Deps,
+    output_type=Response,
+    toolsets=[
+        FunctionToolset[Deps](tools=[get_country], id='country'),
+        MCPServerStdio('python', ['-m', 'tests.mcp_server'], timeout=20, id='mcp'),
+    ],
+    tools=[get_weather],
+    event_stream_handler=event_stream_handler,
+)
+
+# This needs to be done before the `agent` is bound to the workflow.
+temporal_agent = TemporalAgent(
+    agent,
+    activity_config=ActivityConfig(start_to_close_timeout=timedelta(seconds=60)),
+    toolset_activity_config={
+        'country': ActivityConfig(start_to_close_timeout=timedelta(seconds=120)),
+    },
+    tool_activity_config={
+        'country': {
+            'get_country': False,
+        },
+    },
+    run_context_type=TemporalRunContextWithDeps,
+)
+
+
+@workflow.defn
+class AgentWorkflow:
+    @workflow.run
+    async def run(self, prompt: str, deps: Deps) -> Response:
+        result = await temporal_agent.run(prompt, deps=deps)
+        return result.output
+
+
+async def test_temporal(allow_model_requests: None, client: Client, capfire: CaptureLogfire):
+    task_queue = 'pydantic-ai-agent-task-queue'
+
+    async with Worker(
+        client,
+        task_queue=task_queue,
+        workflows=[AgentWorkflow],
+        plugins=[AgentPlugin(temporal_agent)],
+    ):
+        output = await client.execute_workflow(  # pyright: ignore[reportUnknownMemberType]
+            AgentWorkflow.run,
+            args=[
+                'Tell me: the capital of the country; the weather there; the product name',
+                Deps(country='Mexico'),
+            ],
+            id='pydantic-ai-agent-workflow',
+            task_queue=task_queue,
+        )
+        assert output == snapshot(
+            Response(
+                answers=[
+                    Answer(label='Capital of the country', answer='Mexico City'),
+                    Answer(label='Weather in the capital', answer='Sunny'),
+                    Answer(label='Product Name', answer='Pydantic AI'),
+                ]
+            )
+        )
+    exporter = capfire.exporter
+
+    parsed_spans: list[str | AgentStreamEvent | HandleResponseEvent] = []
```

Comment:
```suggestion
    parsed_otel_items: list[str | AgentStreamEvent | HandleResponseEvent] = []
```
or something, since they aren't really spans